✨ Feat: 유저 검색 API

Ko-udon opened the pull request:
## 개요
- 기능 구현
  - 사용자의 닉네임으로 검색 가능
  - 닉네임에 단어가 포함되어 있는 유저 검색 (닉네임: "ko_udon" 검색어: "ko" 검색 가능)
  - icontains로 filter처리

- 비고
  - 테스트 수행 완료

Resolves: #13 

## PR 유형
어떤 변경 사항이 있나요?

- [x] 새로운 기능 추가
- [x] 테스트 추가, 테스트 리팩토링


## PR Checklist
- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다. 
- [x] 변경 사항에 대한 테스트를 했습니다.


![image](https://github.com/HotCake-Tanghuru/HotCake-BE/assets/79897135/31131248-5a8b-4864-8834-fd458fe57213)
(정렬 전)

![image](https://github.com/HotCake-Tanghuru/HotCake-BE/assets/79897135/20c26923-5b73-4190-a512-8f87e3d1a5b8)
(정렬 후)

![image](https://github.com/HotCake-Tanghuru/HotCake-BE/assets/79897135/8ae381c2-abba-4cdc-896c-5be018d4ae1b)
(정렬 전)

![image](https://github.com/HotCake-Tanghuru/HotCake-BE/assets/79897135/ac132d0b-88f4-4823-a525-3551452befb5)
(정렬 후)

Ko-udon committed:
✨ Feat: 유저 검색 API

- 기능 구현
  - 사용자의 닉네임으로 검색 가능
  - 닉네임에 단어가 포함되어 있는 유저 검색 (닉네임: "ko_udon" 검색어: "ko" 검색 가능)
  - icontains로 filter처리

- 비고
  - 테스트 수행 완료

Ko-udon committed:
♻️ Refactor: 검색 결과 정렬

- order_by()로 닉네임 알파벳, 가나다순으로 정렬하여 결과 반환

nayeongdev commented on accounts/urls.py:
```diff
@@ -12,9 +12,10 @@
-    path("users/<int:user_id>/followers", FollowerView.as_view(), name="followers"),
+    path("users/<int:user_id>/following", views.FollowingView.as_view(), name="following"),
+    path("users/<int:user_id>/followers", views.FollowerView.as_view(), name="followers"),
+    path("users/<str:nickname>", views.UserSearch.as_view(), name="follow"),
```
보통 검색할때 /users/search?nickname={query} 이런 형식을 많이 쓰지 않나요??
잘 몰라서 물어보는 겁니다..!

Ko-udon replied:
앗 그럼 혹시 /users/search/<str:nickname> 이런식으로 바꾸면 될까요?

nayeongdev replied:
/users/search/로 하고 파라미터를 views에서 처리하면 될 것 같은데 잘 모르겠네요..ㅠㅠ

Ko-udon replied:
넵 한번 수정해보겠습니다!

nayeongdev replied:
이런식으로 작성할 수 있다고 합니다!
```py
from rest_framework import generics
from rest_framework import filters
from django.contrib.auth.models import User
from .serializers import UserSerializer

class UserSearchView(generics.ListAPIView):
    search_fields = ['username', 'email']
    filter_backends = (filters.SearchFilter,)
    queryset = User.objects.all()
    serializer_class = UserSerializer
```

Ko-udon replied:
수정하였습니다! 감사합니다 한 수 배워갑니다 ㅎㅎ

Ko-udon committed:
♻️ Refactor: 유저 검색 url 구조 변경

- /users/search?nickname={nickname} 으로 변경
- 닉네임 값을 nickname = request.query_params.get("nickname")로 가져오도록 수정
